crates/services/shared-sequencer/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+        let ss_chain_id = http_api::chain_id(&endpoints.blockchain_rest_api)
+            .await?
+            .parse()
+            .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+        let ss_config = http_api::config(&endpoints.blockchain_rest_api).await?;
+
+        let mut minimum_gas_price = ss_config.minimum_gas_price;
+
+        if let Some(index) = minimum_gas_price.find('.') {
+            minimum_gas_price.truncate(index);
+        }
+        let gas_price: u128 = minimum_gas_price.parse()?;
+        // Ceil the gas price to the next integer.
+        let gas_price = gas_price.saturating_add(1);
+
+        Ok(Self {
+            topic,
+            endpoints,
+            account_prefix,
+            coin_denom,
+            ss_chain_id,
+            gas_price,
+        })
+    }
+
+    /// Returns the Cosmos account ID of the sender.
+    pub fn sender_account_id<S: Signer>(&self, signer: &S) -> anyhow::Result<AccountId> {
+        let sender_public_key = signer.public_key();
+        let sender_account_id = sender_public_key
+            .account_id(&self.account_prefix)
+            .map_err(|err| anyhow!("{err:?}"))?;
+
+        Ok(sender_account_id)
+    }
+
+    fn tendermint(&self) -> anyhow::Result<tendermint_rpc::HttpClient> {
+        Ok(tendermint_rpc::HttpClient::new(
+            &*self.endpoints.tendermint_rpc_api,
+        )?)
+    }
+
+    /// Retrieve latest block height
+    pub async fn latest_block_height(&self) -> anyhow::Result<u32> {
+        Ok(self
+            .tendermint()?
+            .abci_info()
+            .await?
+            .last_block_height
+            .value()
+            .try_into()?)
+    }
+
+    /// Retrieve account metadata by its ID
+    pub async fn get_account_meta<S: Signer>(
+        &self,
+        signer: &S,
+    ) -> anyhow::Result<AccountMetadata> {
+        let sender_account_id = self.sender_account_id(signer)?;
+        http_api::get_account(&self.endpoints.blockchain_rest_api, sender_account_id)
+            .await
+    }
+
+    /// Retrieve the topic info, if it exists
+    pub async fn get_topic(&self) -> anyhow::Result<Option<TopicInfo>> {
+        http_api::get_topic(&self.endpoints.blockchain_rest_api, self.topic).await
+    }
+
+    /// Post a sealed block to the sequencer chain using some
+    /// reasonable defaults and the config.
+    /// This is a convenience wrapper for `send_raw`.
+    pub async fn send<S: Signer>(
+        &self,
+        signer: &S,
+        account: AccountMetadata,
+        order: u64,
+        blob: Vec<u8>,
+    ) -> anyhow::Result<()> {
+        let latest_height = self.latest_block_height().await?;
+
+        self.send_raw(
+            // We don't want our transactions to be stay in the mempool for a long time,
+            // so we set the timeout height to be 64 blocks ahead of the latest block height.
+            // The 64 is an arbitrary number.
+            latest_height.saturating_add(64),
```

Comment:
Do we know at what pace the sequencer chain produces blocks?

Reply:
Once per second, IIRC

Reply:
SS network produces block each 6 seconds